File: String/string_same_length.cxx
/* AtCoder Beginner Contest 148 (Task B) */


#include <bits/stdc++.h>
using namespace std;

class Solution{
    public:
    static string str(string &s1, string &s2){
        string res = "";
        for (auto i = 0; s1[i]&&s2[i]; ++i){
            res += s1[i];
            res += s2[i];
        }
        return res;
    }
};
int main(){
    ios_base::sync_with_stdio(0);
    string s1 = "ip";
    string s2 = "cc";
    cout << Solution::str(s1,s2); // icpc
}

File: String/string_same_length.py
# AtCoder Beginner Contest 148 (Task B)

class Solution(object):
    def str(self,s1:str,s2:str)->str:
        self.res = ''
        if(len(s1)!=len(s2)):
            return -1
        for i in range(len(s1)):
            self.res += s1[i]
            self.res += s2[i]
        return self.res



if __name__=='__main__':
    s1 = "ip"
    s2 = "cc"
    s = Solution()
    print(s.str(s1,s2)) # icpc
